### PR TITLE
Adjusted Ethereum Tests to Cancun + No 4788 Execution if no account (contract not deployed)

### PR DESCRIPTION
**Pull request by MarekM25**
## Changes

- Adjusted Ethereum Tests to Cancun
- no 4788 execution in case of no account
- Updated tests to the latest version
- Small cleanups

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [x] Other: Regular update of Ethereum tests

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Requires documentation update

- [ ] Yes
- [x] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs [hunk 1]
```diff
@@ -80,13 +80,21 @@ protected EthereumTestResult RunTest(GeneralStateTest test, ITxTracer txTracer)
...
+                test.CurrentNumber,
+                test.CurrentGasLimit,
+                test.CurrentTimestamp,
+                Array.Empty<byte>());
             header.BaseFeePerGas = test.Fork.IsEip1559Enabled ? test.CurrentBaseFee ?? _defaultBaseFeeForStateTest : UInt256.Zero;
             header.StateRoot = test.PostHash;
             header.Hash = header.CalculateHash();
             header.IsPostMerge = test.CurrentRandom is not null;
             header.MixHash = test.CurrentRandom;
+            header.WithdrawalsRoot = test.CurrentWithdrawalsRoot;
```
Comment (emlautarom1) [inline src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:97]: What do we do with:

- `test.CurrentBeaconRoot`
- `test.ParentBlobGasUsed`
- `test.ParentExcessBlobGas`

Comment (emlautarom1) [reply src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:97]: I've tried adding the following lines:

```C#
header.ParentBeaconBlockRoot = test.CurrentBeaconRoot;
// ...
if (test.ParentBlobGasUsed is not null && test.ParentExcessBlobGas is not null)
{
    BlockHeader parent = new(
        parentHash: null!,
        unclesHash: Keccak.OfAnEmptySequenceRlp,
        beneficiary: test.CurrentCoinbase,
        difficulty: test.CurrentDifficulty,
        number: test.CurrentNumber - 1,
        gasLimit: test.CurrentGasLimit,
        timestamp: test.CurrentTimestamp,
        extraData: Array.Empty<byte>()
    )
    {
        BlobGasUsed = (ulong)test.ParentBlobGasUsed,
        ExcessBlobGas = (ulong)test.ParentExcessBlobGas,
    };
    header.ExcessBlobGas = BlobGasCalculator.CalculateExcessBlobGas(parent, spec);
}
```

The intent was to do the same thing as geth is doing:

https://github.com/ethereum/go-ethereum/blob/e9f59b5d5ea44df607f826c72f42916b552ab33d/cmd/evm/internal/t8ntool/execution.go#L167-L180

Unfortunately, this does not produce any observable change to the failing tests: adding or not these lines doesn't change the actual state root (`0xfb267ebc2b5a092a18b6e3828b4ae2c57ece236db0b5afcdab0ddd233ab1d983`), which is not the expected one (`0x52196f6444dac6b01163ab7f09586ce9b5a961c271708b5a3fb3d5a862266722`).

Maybe this extra data is irrelevant and the issue is somewhere else?

Comment (emlautarom1) [reply src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:97]: Fixed: we were not properly deserializing the `Transaction` (missing fields) and we were computing the wrong `TxType`.

File: src/Nethermind/Ethereum.Blockchain.Test/BadOpcodeTests.cs [hunk 2]
```diff
@@ -21,7 +22,7 @@ public void Test(GeneralStateTest test)
         public static IEnumerable<GeneralStateTest> LoadTests()
         {
             var loader = new TestsSourceLoader(new LoadGeneralStateTestsStrategy(), "stBadOpcode");
-            return (IEnumerable<GeneralStateTest>)loader.LoadTests();
+            return ((IEnumerable<GeneralStateTest>)loader.LoadTests());
```
Comment (LukaszRozmej) [inline src/Nethermind/Ethereum.Blockchain.Test/BadOpcodeTests.cs:25]: minor - unneeded parenthesis

File: src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs [hunk 3]
```diff
@@ -80,22 +80,49 @@ protected EthereumTestResult RunTest(GeneralStateTest test, ITxTracer txTracer)
...
 
             Stopwatch stopwatch = Stopwatch.StartNew();
             TxValidator? txValidator = new((MainnetSpecProvider.Instance.ChainId));
             IReleaseSpec? spec = specProvider.GetSpec((ForkActivation)test.CurrentNumber);
             if (test.Transaction.ChainId == null)
                 test.Transaction.ChainId = MainnetSpecProvider.Instance.ChainId;
+            if (test.ParentBlobGasUsed is not null && test.ParentExcessBlobGas is not null)
+            {
+                BlockHeader parent = new(
+                    parentHash: null!,
```
Comment (LukaszRozmej) [inline src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:108]: parent hash can be null?

**Review by LukaszRozmej — APPROVED** [review]

**Review by marcindsobczak — APPROVED** [review]

**Pull request merged**